File: problem-002/nim/efibsum.nim
proc efibsum(n: int, nprev: int, max: int): int = 
    if n > max:
        result = 0
    elif n.mod(2) != 0:
        result = efibsum(n + nprev, n, max)
    else:
        result = n + efibsum(n + nprev, n, max)

echo(efibsum(1, 1, 4000000))

File: problem-002/elixir/efibsum.exs
defmodule P002 do
  def efibsum(n, nprev, max) do
    cond do
      n > max -> 0
      rem(n, 2) != 0 -> efibsum(n+nprev, n, max)
      true -> n + efibsum(n+nprev, n, max)
    end
  end
end

IO.puts(P002.efibsum(1, 1, 4000000))
